Source/Shaders/Builtin/Functions/discardIfClipped.glsl

```diff
@@ -1,36 +1,47 @@
 /**
- * Clip a fragment by an array of clipping planes.
+ * Clip a fragment by an array of clipping planes. Clipping plane regions are not combined, therefore if a fragment is
+ * clipped by any of the planes, it is discarded.
  *
  * @name czm_discardIfClipped
  * @glslFunction
  *
  * @param {vec4[]} clippingPlanes The array of planes used to clip, defined in eyespace.
  * @param {int} clippingPlanesLength The number of planes in the array of clipping planes.
+ * @returns {float} The distance away from a clipped fragment, in eyespace
  */
-void czm_discardIfClipped (vec4[czm_maxClippingPlanes] clippingPlanes, int clippingPlanesLength)
+float czm_discardIfClipped (vec4[czm_maxClippingPlanes] clippingPlanes, int clippingPlanesLength)
 {
     if (clippingPlanesLength > 0)
     {
-        bool clipped = false;
         vec4 position = czm_windowToEyeCoordinates(gl_FragCoord);
         vec3 clipNormal = vec3(0.0);
         vec3 clipPosition = vec3(0.0);
+        float clipAmount = 0.0;
+        float pixelWidth = czm_metersPerPixel(position);
 
         for (int i = 0; i < czm_maxClippingPlanes; ++i)
         {
-            if (i == clippingPlanesLength || clipped)
+            if (i == clippingPlanesLength)
             {
                 break;
             }
 
             clipNormal = clippingPlanes[i].xyz;
             clipPosition = -clippingPlanes[i].w * clipNormal;
-            clipped = dot(clipNormal, (position.xyz - clipPosition)) <= 0.0;
-        }
 
-        if (clipped)
-        {
-            discard;
+            float amount = dot(clipNormal, (position.xyz - clipPosition)) / pixelWidth;
+            if (amount > clipAmount) {
+                clipAmount = amount;
+            }
```

Comment:
The compiler my optimize this `if` out, but a faster version would be `clipAmount = max(amount, clipAmount)`